src/formatting/report.rs

```diff
@@ -154,6 +158,16 @@ impl FormatReport {
             .insert(format_error);
     }
 
+    fn has_error_kind(&self, kind: ErrorKind) -> bool {
+        RefCell::borrow(&self.format_result)
+            .iter()
+            .any(|(_, format_result)| format_result.has_error_kind(kind))
+    }
+
+    pub fn has_invalid_rustfmt_attribute_errors(&self) -> bool {
+        self.has_error_kind(ErrorKind::BadAttr)
+    }
+
```

Comment:
I don't have any other specific scenarios in mind where consumers of the rustfmt lib may need insight into the specific formatting error kinds, but went ahead and implemented the immediate need in a way that should be reusable should the need arise.